README.md

```diff
@@ -1,3 +1,197 @@
+Some other, like us at M6Web, prefers to use a less granular one and to manage consents at a purpose level.
+
+**Example of a vendor based consent management:**
+
+- [x] Google Analytics
+- [ ] Krux
+
+**Example of a vendor based consent management:**
+
+- [ ] Information storage and access
+- [x] Personalisation
+
+_Feature based is not available for now with IAB Consent Framework. But since the feature field is available in the VendorList, we can imagine managing consents using feature based approach_
+
+## Consent Management Provider (CMP)
+
+"CMP" means a company that can read the vendors chosen by a website operator and the consent status of an end user (either service specific (through a first-party cookie) or global (through a third-party cookie).
+
+### Required API
+
+Every consent manager MUST provide the following API:
+
+`__cmp(Command, Parameter, Callback)`
+
+This API provide few command : 
+
+<table>
+  <tr>
+    <td>Command</td>
+    <td>Parameter</td>
+    <td>Callback</td>
+    <td>Comments</td>
+  </tr>
+  <tr>
+    <td>getVendorConsents</td>
+    <td>Vendors Ids [Array:String]</td>
+    <td>Callback(VendorConsents object, success: boolean)</td>
+    <td>Vendor consents strings</td>
+  </tr>
+  <tr>
+    <td>getConsentData</td>
+    <td>Consent string [String]</td>
+    <td>Callback(VendorConsentData object, success: boolean)</td>
+    <td>Consents data from user</td>
+  </tr>
+  <tr>
+    <td>ping</td>
+    <td> - </td>
+    <td>Callback(PingReturn object, success: boolean)</td>
+    <td>The "ping" command invokes the callback immediately with information about whether the main CMP script has loaded yet and if GDPR has been configured for all users or just EU users. </td>
+  </tr>
+</table>
+
+### Communication with vendors
+
+For communication with vendors we mostly use the `__cmp` function for local communication and via `iframe` (standard).
+
+#### `window.__cmp`
+The function __cmp is created like this : 
+```js
+export const __cmp = {
+  getVendorConsents: vendorConsentsLoader,
+  getConsentData: vendorConsentDataLoader,
+  ping: () => {
+    const isCmpLoaded = !!document.querySelector('iframe[name=__cmpLocator]');
+
+    return new PingReturn(isCmpLoaded);
+  },
+};
+```
+
+So you can call this function anywhere in the code.
+
+For exemple, you cann get vendor consents :
+
+```js
+__cmp.getVendorConsents()
+```
+
+#### Via (standard) iframe
+
+Standard iframe is call with eventListener in this function : 
+```js
+export const bootIabStandardIframe = () => {
+  window.addEventListener('message', handleMessage);
+
+  const iframe = document.createElement('iframe');
+  iframe.name = '__cmpLocator';
+  iframe.width = 0;
+  iframe.height = 0;
+  iframe.style.display = 'none';
+  document.body.appendChild(iframe);
+};
+```
+
+if a message arrive with `data.__cmpCall` the iframe is call and user can choose his data feed with us.
```

Comment:
In fact, the iframe always exist since **the user has given access to its data**

And then, the app (6Play in our case) is able to "chat" with its child frame.

The child frames will send messages to 6Play that will return information concerning the previously agreed consents.

WYT of rephrasing just a bit this part?